### PR TITLE
Adding Project Menu to Play

### PR DESCRIPTION
**Pull request by wu-emi**
## 💸 TL;DR
Changing "Export" in the top bar into a drop-down "Project" menu to add "Save" and "Load" to?

## 📜 Details
Screenshot 2024-08-20 at 3 09 34 PM

## ✅ Checks

- [ ] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: src/elements/play-project-button.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                size=${iconSizes[this.size]}
+                icon="caret-down-outline"
+              ></play-icon>
+            </button>
+          </div>
+          <div slot="menu">
+            <play-list-item
+              label="Save"
+              icon="download-outline"
+            ></play-list-item>
```
Comment (gunsch) [inline src/elements/play-project-button.ts:177]: we should have "Save", "Load", and "Export"

and maybe have Save and Load commented out for now until we wire them up? but the code can go ahead and be here :)

File: src/elements/play-pen-header.ts [hunk 2]
```diff
@@ -118,9 +117,6 @@ export class PlayPenHeader extends LitElement {
   @query('play-assets-dialog')
   private _assets!: PlayAssetsDialog
 
-  @query('play-export-dialog')
-  private _export!: PlayExportDialog
```
Comment (gunsch) [inline src/elements/play-pen-header.ts:122]: I'm confused --- where did this functionality go? this should still be what "Export" does :)

Comment (wu-emi) [reply src/elements/play-pen-header.ts:122]: Oops I thought you wanted to replace it! I'll add it back in as an option in the dropdown menu with the dialog!

**Comment by wu-emi** [comment]
image
Updated screenshot!

File: src/elements/play-pen-header.ts [hunk 3]
```diff
@@ -118,12 +118,15 @@ export class PlayPenHeader extends LitElement {
...
-
   @query('play-settings-dialog')
   private _settings!: PlaySettingsDialog
 
+  protected override firstUpdated(): void {
+    this.addEventListener('open-export-dialog', () => {
+      this.shadowRoot?.querySelector('play-export-dialog')?.open()
```
Comment (gunsch) [inline src/elements/play-pen-header.ts:126]: curious why this instead of `this._export.open()` like was here before?

**Review by gunsch — APPROVED** [review]

**Pull request merged**